dnd-troops-web/src/main/java/cz/muni/fi/pa165/w2018/dndtroops/web/controllers/RoleController.java

```diff
@@ -0,0 +1,177 @@
+package cz.muni.fi.pa165.w2018.dndtroops.web.controllers;
+
+import cz.muni.fi.pa165.w2018.dndtroops.api.dto.RoleChangeDTO;
+import cz.muni.fi.pa165.w2018.dndtroops.api.dto.RoleDTO;
+import cz.muni.fi.pa165.w2018.dndtroops.api.dto.SearchNameDTO;
+import cz.muni.fi.pa165.w2018.dndtroops.api.facade.RoleFacade;
+import cz.muni.fi.pa165.w2018.dndtroops.web.WebUris;
+import cz.muni.fi.pa165.w2018.dndtroops.web.helpers.FlashMessage;
+import cz.muni.fi.pa165.w2018.dndtroops.web.helpers.FlashMessageType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import javax.validation.Valid;
+import java.util.List;
+
+/**
+ * Class RoleController
+ *
+ * @author Marek Valko
+ */
+@Controller
+@RequestMapping(value = {WebUris.URL_ROLE})
+public class RoleController extends BaseController {
+    private final static Logger log = LoggerFactory.getLogger(RoleController.class);
+
+    @Autowired
+    private RoleFacade roleFacade;
+
+
+    @RequestMapping(value = "/all", method = RequestMethod.POST)
```

Comment:
Rename it to something like "/search/name" so it's not confusing.
You also need to change it in jsp afterwards

Then you will just in case of bindingResult.hasErrors() or roleDTO == null redirect to handler for /all with flashAttributes in same way as you do in createPost